### PR TITLE
Credit checkout update

### PR DESCRIPTION
**Pull request by clintonb**
- Checking to see if a seat is actually available for purchase, before displaying info to user.
- Displaying a more helpful error message if no seat is available.

ECOM-4378

### CODE REVIEW

**Comment by clintonb** [comment]
@rlucioni please review

File: ecommerce/credit/tests/test_views.py [hunk 1]
```diff
@@ -6,13 +6,14 @@
 import json
 
 import ddt
+from datetime import datetime, timedelta
```
Comment (rlucioni) [inline ecommerce/credit/tests/test_views.py]: Nitpick: `datetime` is part of the standard library. pylint>=1.5.4 will catch this.

File: ecommerce/credit/views.py [hunk 2]
```diff
@@ -32,34 +34,48 @@ def get_context_data(self, **kwargs):
         context = super(Checkout, self).get_context_data(**kwargs)
 
         course = get_object_or_404(Course, id=kwargs.get('course_id'))
+        context['course'] = course
```
Comment (rlucioni) [inline ecommerce/credit/views.py:37]: Left over from debugging? This was added but I don't see it being used.

File: ecommerce/credit/tests/test_views.py [hunk 3]
```diff
@@ -285,3 +268,18 @@ def test_disabled_providers(self, disabled_processors, expected_context):
         response = self.client.get(self.path)
         self.assertEqual(response.status_code, 200)
         self.assertDictContainsSubset(expected_context, response.context)
+
+    def test_seat_unavailable(self):
```
Comment (rlucioni) [inline ecommerce/credit/tests/test_views.py:274]: Failures suggest you may need to use activate/use httpretty for this test.

Comment (rlucioni) [inline ecommerce/credit/tests/test_views.py:274]: Looks like you'll at least want to `_mock_eligibility_api`.

Comment (clintonb) [inline ecommerce/credit/tests/test_views.py:274]: Fixed.

File: ecommerce/credit/tests/test_views.py (hunk 1 above)
Comment (clintonb) [inline ecommerce/credit/tests/test_views.py]: Fixed.

File: ecommerce/credit/views.py (hunk 2 above)
Comment (clintonb) [inline ecommerce/credit/views.py:37]: Nope. Previously, the error conditions did not add the course to the context. This resulted in the title of the page being incorrect. This fixes that.

**Comment by rlucioni** [comment]
👍

**Comment by clintonb** [comment]
screen shot 2016-05-05 at 5 58 28 pm

**Pull request merged**